pmm-app/src/pmm-dbaas/components/DBCluster/EditDBClusterModal/EditDBClusterModal.tsx

```diff
@@ -0,0 +1,103 @@
+import { Modal } from 'shared/components/Elements/Modal/Modal';
+import { Form as FormFinal } from 'react-final-form';
+import { EditDBClusterFields, EditDBClusterModalProps } from './EditDBClusterModal.types';
+import { DBClusterAdvancedOptions } from './DBClusterAdvancedOptions/DBClusterAdvancedOptions';
+import { DEFAULT_SIZES } from './DBClusterAdvancedOptions/DBClusterAdvancedOptions.constants';
+import { DBClusterTopology } from './DBClusterAdvancedOptions/DBClusterAdvancedOptions.types';
+import { DBClusterServiceFactory } from '../DBClusterService.factory';
+
+export const EditDBClusterModal: FC<EditDBClusterModalProps> = ({
+  isVisible,
+  setVisible,
+  onDBClusterChanged,
+  selectedCluster,
+}) => {
+  const onSubmit = async ({
+    databaseType,
+    topology,
+    nodes,
+    single,
+    memory,
+    cpu,
+    disk,
+  }: Record<string, any>) => {
+    if (!selectedCluster) {
+      setVisible(false);
+
+      return;
+    }
+
+    try {
+      const dbClusterService = DBClusterServiceFactory.newDBClusterService(databaseType.value);
+
+      await dbClusterService.updateDBCluster({
+        databaseType,
+        clusterName: selectedCluster.clusterName,
+        kubernetesClusterName: selectedCluster.kubernetesClusterName,
+        clusterSize: topology === DBClusterTopology.cluster ? nodes : single,
+        cpu,
+        memory,
+        disk,
+      });
+      setVisible(false);
+      onDBClusterChanged();
+    } catch (e) {
+      console.error(e);
+    }
+  };
+
+  const [initialValues, setInitialValues] = useState({});
+
+  useEffect(() => {
+    if (!selectedCluster) {
+      setVisible(false);
+
+      return;
+    }
+
+    const clusterParameters: any = { ...selectedCluster };
+
+    clusterParameters.databaseType = {
+      value: clusterParameters.databaseType,
+    };
+    // eslint-disable-next-line max-len
+    clusterParameters.topology = clusterParameters.clusterSize > 1 ? DBClusterTopology.cluster : DBClusterTopology.single;
+    clusterParameters[EditDBClusterFields.nodes] = clusterParameters.clusterSize;
+    clusterParameters[EditDBClusterFields.single] = clusterParameters.clusterSize;
+
+    const isMatchSize = (type) => DEFAULT_SIZES[type].cpu === clusterParameters.cpu
+      && DEFAULT_SIZES[type].memory === clusterParameters.memory
+      && DEFAULT_SIZES[type].disk === clusterParameters.disk;
+
+    if (isMatchSize('small')) {
+      clusterParameters.resources = 'small';
+    } else if (isMatchSize('medium')) {
+      clusterParameters.resources = 'medium';
+    } else if (isMatchSize('large')) {
+      clusterParameters.resources = 'large';
+    } else {
+      clusterParameters.resources = 'custom';
+    }
+
+    setInitialValues(clusterParameters);
+  }, [selectedCluster]);
+
+  return (
+    // eslint-disable-next-line max-len
+    <Modal title={`Edit cluster ${selectedCluster?.clusterName} (${selectedCluster?.databaseType})`} isVisible={isVisible} onClose={() => setVisible(false)}>
```

Comment:
Break to different lines and remove the comment to be consistent.
nit: put a function in Messages file to construct this string

Reply:
Fixed, not moved it to the message because I'm not sure if we ever did that and don't want to overload Messages file